client/views/room/Header/QuickActions/QuickActions.tsx

```diff
@@ -0,0 +1,207 @@
+		closeModal();
+		if (err != null) {
+			return handleError(err);
+		}
+
+		toastr.success(t('Your_email_has_been_queued_for_sending'));
+	});
+
+	const discardTranscript = (): void => Meteor.call('livechat:discardTranscript', rid, (error: any) => {
+		closeModal();
+		if (error != null) {
+			return handleError(error);
+		}
+		toastr.success(t('Livechat_transcript_request_has_been_canceled'));
+	});
+
+	const forwardChat = (departmentId?: string, userId?: string, comment?: string): void => {
+		if (departmentId && userId) {
+			return;
+		}
+		const transferData: { roomId: string; comment?: string; departmentId?: string; userId?: string } = {
+			roomId: rid,
+			comment,
+		};
+
+
+		if (departmentId) { transferData.departmentId = departmentId; }
+		if (userId) { transferData.userId = userId; }
+
+		Meteor.call('livechat:transfer', transferData, (error: any, result: any) => {
+			if (error) {
+				toastr.error(t(error.error));
+			} else if (result) {
+				closeModal();
+				toastr.success(t('Transferred'));
+				FlowRouter.go('/');
+			} else {
+				toastr.warning(t('No_available_agents_to_transfer'));
+			}
+		});
+	};
+
+	const confirmClose = (comment: string): void => Meteor.call('livechat:closeRoom', rid, comment, { clientAction: true }, (error: any) => {
+		closeModal();
+		if (error) {
+			return handleError(error);
+		}
+
+		toastr.success(t('Chat_closed_successfully'));
+	});
+
+	const openModal = useMutableCallback((id: string) => {
+		switch (id) {
+			case QuickActionsEnum.MoveQueue:
+				setModal(<ReturnChatQueueModal onMoveChat={moveChat} onCancel={closeModal} />);
+				break;
+			case QuickActionsEnum.Transcript:
+				setModal(<TranscriptModal room={room} email={email} onRequest={requestTranscript} onSend={sendTranscript} onDiscard={discardTranscript} onCancel={closeModal} />);
+				break;
+			case QuickActionsEnum.ChatForward:
+				setModal(<ForwardChatModal onForward={forwardChat} onCancel={closeModal} />);
+				break;
+			case QuickActionsEnum.CloseChat:
+				setModal(<CloseChatModal onConfirm={confirmClose} onCancel={closeModal} />);
+				break;
+			default:
+				break;
+		}
+	});
+
+	const actionDefault = useMutableCallback((e) => {
+		const index = e.currentTarget.getAttribute('data-quick-actions');
+		const { id } = actions[index];
+		openModal(id);
+	});
+
+	const hasManagerRole = useRole('livechat-manager');
+
+	const roomOpen = room && room.open && ((room.servedBy && room.servedBy._id === uid) || hasManagerRole);
+
+	const canForwardGuest = usePermission('transfer-livechat-guest');
+
+	const canSendTranscript = usePermission('send-omnichannel-chat-transcript');
+
+	const canCloseRoom = usePermission('close-others-livechat-room');
+
+	const canReturnQueue = useOmnichannel().routeConfig?.returnQueue;
+
+	const hasPermissionButtons = (id: string): boolean => {
```

Comment:
Permissions are not working if the user is not a `livechat-manager`.

Reply:
https://user-images.githubusercontent.com/2493803/111843736-77ddab80-88e0-11eb-8f12-2e12ce75707f.mp4

Reply:
Is the user `admin` also?

Reply:
![image](https://user-images.githubusercontent.com/2493803/111846115-ec1a4e00-88e4-11eb-801f-0473c41f69c6.png)

weird, same here

Reply:
The point is: How can your test return `true` if the room Object doesn't return the `servedBy` property?
Can you confirm you get this property on your end?

Reply:
Yes, Isn't coming from `RoomProvider`, I need to look deeply.

Reply:
It might be something related to Interface / type casting.
But the information needs to be retrieved from the DB also.
https://github.com/RocketChat/Rocket.Chat/blob/develop/definition/IRoom.ts#L50-L55